ext/liquid_c/vm_assembler.h

```diff
@@ -101,8 +102,12 @@ static inline void vm_assembler_add_filter(vm_assembler_t *code, VALUE filter_na
 
 static inline void vm_assembler_add_render_variable_rescue(vm_assembler_t *code, size_t node_line_number)
 {
-    c_buffer_write(&code->constants, &node_line_number, sizeof(size_t));
-    vm_assembler_write_opcode(code, OP_RENDER_VARIABLE_RESCUE);
+    if (RB_UNLIKELY(node_line_number >= (1 << 24))) {
+        rb_enc_raise(utf8_encoding, rb_eRuntimeError, "Line number %ld is too large", node_line_number);
```

Comment:
I think `ArgumentError` (rb_eArgError) would be more appropriate existing exception class for this.

"Line number too large 16777217" might be a confusing error message.  It seems like an error message that makes sense to the caller of `vm_assembler_add_render_variable_rescue`, but this is a ruby exception being presented to the caller of `Liquid::Template.parse`.  It might be clearer if the error message was related to the input to that.

Should we instead check the source code length?  That way we could easily check it upfront instead of throughout parsing.  We could also state more clearly that only a template source of up to 16 MiB is supported.  That would also allow us to use that assumption elsewhere.